src/actions/Account.actions.ts

```diff
@@ -1,56 +1,68 @@
-import { ethers } from "ethers";
+import { BigNumber, ethers } from "ethers";
 import { addresses, Actions } from "../constants";
 import { abi as ierc20Abi } from "../abi/IERC20.json";
 import { abi as sOHM } from "../abi/sOHM.json";
 import { abi as sOHMv2 } from "../abi/sOhmv2.json";
+import { Dispatch } from "redux";
+import { StaticJsonRpcProvider } from "@ethersproject/providers";
+import { IERC20 } from "src/typechain";
 
-export const fetchAccountSuccess = payload => ({
+interface IAccountDetails {
+  readonly balances: { [token: string]: string };
+  readonly bonding: { daiAllowance: number | BigNumber };
+  readonly migrate: { unstakeAllowance: BigNumber | undefined };
+  readonly staking: { ohmStake: number | BigNumber; ohmUnstake: number | BigNumber };
+}
+export const fetchAccountSuccess = (payload: IAccountDetails) => ({
   type: Actions.FETCH_ACCOUNT_SUCCESS,
   payload,
 });
 
 export const loadAccountDetails =
-  ({ networkID, provider, address }) =>
-  async dispatch => {
+  ({ networkID, provider, address }: { networkID: number; provider: StaticJsonRpcProvider; address: string }) =>
+  async (dispatch: Dispatch) => {
     // console.log("networkID = ", networkID)
     // console.log("addresses = ",addresses)
 
-    let ohmBalance = 0;
-    let sohmBalance = 0;
-    let oldsohmBalance = 0;
-    let stakeAllowance = 0;
-    let unstakeAllowance = 0;
+    let ohmBalance: number | BigNumber = 0;
```

Comment:
because we are initially giving it a type of number (= 0) and then setting it to type `BigNumber` w/ `await ohmContract.balanceOf(address)`. but I have changed it to a more suitable singular type.